conda/resolve.py

```diff
@@ -7,8 +7,8 @@
 import copy
 from logging import DEBUG, getLogger
 
-from .auxlib.collection import frozendict
 from .auxlib.decorators import memoize, memoizemethod
+from ._vendor.frozendict import frozendict
```

Comment:
If I understand this correctly, the vendored frozendict library doesn't use `collections.OrderedDict` as the one in auxlib but an `collections.abc.Mapping` as its base class, which I think (!) means that it's not going to be ordered by default anymore. This is probably going to have a negative impact on the `Resolve.__hash__` method since it won't changes the hash of the various values used with `frozendict`. Can we double-check if `collections.abc.Mapping` are ordered now on Python >= 3.7?

Reply:
Actually, I don't think this makes a difference anymore given that we are [dropping Python 3.6 support](https://github.com/conda/conda/issues/11358). While `frozendict` inherits from `Mapping` it's using a private, internal `dict` instance:
https://github.com/conda/conda/blob/077616b2f3f19eae332613d304e68f4ca3caae76/conda/_vendor/frozendict/__init__.py#L18-L21

So both `frozendict` and `FrozenOrderedDict` are actually ordered implementations at this time.